### DISCUSSION TITLE
Multiple replicas of metric server

### DISCUSSION THREAD

**Discussion by yuvalweber** [kedacore/keda-docs#997] [General]
I had question regarding metricServer high-availability.
In your operational doc you said that metric-server supports multiple replicas and you suggest that but you said that the number of replicas is only 1.
Can you explain if there is a support in multiple metricsServer and if there is something to take in consideration when doing so, and if so what is the number of replicas recommended?

**Comment by tomkerkhove** [comment]
You can run it with multiple replicas but it's more of an active passive scenario and not a load-balancing solution

**Reply by yuvalweber** [reply]
hey @tomkerkhove are you sure? Cause based on what written on your documentation you said that metric-server can work like load-balancing if enabling some flag in apiserver

**Reply by tomkerkhove** [reply]
Right, I might be wrong here - @zroubalik? Does this change with our new metric scraping approach? Because if that is correct then it might be able to fan out then?

**Reply by zroubalik** [reply]
That's up to Kubernetes configuration, usually the behavior is like @tomkerkhove described, if you enable the setting on k8s, the requests are routed across all replicas. Though I am not sure it benefitical enough, most of the the requests coming to KEDA Metrics Server are being forwarded to KEDA operator anyway.

This doesn't change with our new metrics scraping approach.

**Reply by yuvalweber** [reply]
alright thanks.
Just wondering, isn't keda operator only gets the metric if it's a task that related only to him (like scaling to 0)? @zroubalik

**Reply by zroubalik** [reply]
See this issue: https://github.com/kedacore/keda/issues/2282

**Reply by yuvalweber** [reply]
I read this issue and I have one final question cause I'm not sure I totally understand.
Regarding **pollingInterval** you wrote in this issue that it queries the external scaler every time the k8s-server asks him to, but isn't the k8s-server asks him to query the external scaler only based on the pollingInteval.
Like what other cases does the kubernetes-server would ask him for metric? @zroubalik

**Reply by zroubalik** [reply]
pollingInterval is only for `0<->1` scaling. See this for reference: https://keda.sh/docs/2.9/concepts/scaling-deployments/#caching-metrics-experimental

**Reply by yuvalweber** [reply]
@zroubalik but regarding what I read from your answer in the issue it seems like regarding this caching mechanism the pollingInterval would work also for scaling as it is. No?

**Reply by zroubalik** [reply]
If you enable the new feature then yes.
